src/cargo/util/config.rs

```diff
@@ -70,6 +70,8 @@ pub struct Config {
     cache_rustc_info: bool,
     /// Creation time of this config, used to output the total build time
     creation_time: Instant,
+    /// Target Directory via resolved Cli parameter 
+    cli_target_dir: Option<Filesystem>,
```

Comment:
I think that instead of `cli_target_dir`, we can store computed `target_dir`. That is, we can change `fn target_dir()` to

```
fn target_dir(&self) -> Option<FileSystem> { self.target_dir.clone() }
```

and instead move all logic from it to `configure` method. I think that would be slightly more clear, because we'll only *store* something, instead of both storing and computing on the fly. What do you think?

Reply:
Agreed! 

I had some problems having the logic in `configure` as it would change some error messages with bad configs. But I was able to minimize it to just a `bad_config::invalid_global_config` situation, so it should be good.